src/tun2proxy.rs

```diff
@@ -274,6 +278,10 @@ impl<'a> TunToProxy<'a> {
             Interest::READABLE,
         )?;
 
+        let mut _exit_listener = mio::net::TcpListener::bind(EXIT_LISTENER.parse()?)?;
```

Comment:
In the long term, we should probably use a pipe here: [mio_pipe](https://docs.rs/mio-pipe/latest/mio_pipe/index.html)::[new_pipe](https://docs.rs/mio-pipe/latest/mio_pipe/fn.new_pipe.html)

The current implementation would allow any application on localhost to initiate a shutdown.